deploy/charts/alluxio/templates/jobs/load-script.yaml

```diff
@@ -0,0 +1,51 @@
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+{{- $name := include "alluxio.name" . }}
+{{- $fullName := include "alluxio.fullname" . }}
+{{- $chart := include "alluxio.chart" . }}
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $fullName }}-alluxio-load
+  labels:
+    name: {{ $fullName }}-alluxio-load
+    app: {{ $name }}
+    chart: {{ $chart }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  load.sh: |-
+    #!/usr/bin/env bash
+
+    main() {
+      if [[ $# != 1 ]]; then
+        echo "Load command only takes one argument"
+        exit 1
+      fi
+      exec /opt/alluxio/bin/alluxio fs load $1 --submit
+      waitUntilLoadFinished
+    }
+
+    waitUntilLoadFinished() {
+      while true; do {
```

Comment:
would like to suggest adding a timeout scenario, in case the progress command stalls or never returns one of the expected cases. i commonly use the ticker + timeout switch statement in go to do this efficiently; possible write this outside of bash?

Reply:
Outside of bash you mean, for example, use go instead?

Reply:
yah can this call `go run` and point to a main file? i guess it would depends if the env has go installed...

Reply:
Or create a new image based on alluxio/alluxio image that installs go...

Reply:
yeah i think it'll be better if there would be a timeout. if the golang is not available, i think we can simply use `seq` with `for`. it's an idiom i used to use when you don't need the time to be very accurate.

```shell
WAIT_RETRY_TIMES=40 # approximately 10min

for i in $(seq ${WAIT_RETRY_TIMES}); do
  sleep 15
  res=$(/opt/alluxio/bin/alluxio fs load $1 --progress)
  if [[ "${res}" == *"SUCCEEDED"* ]]; then
    exit 0
  elif [[ "${res}" == *"SUCCEEDED"* ]]; then
    exit 1
  fi
done
```

Reply:
I think what Rico was saying is the stall of the `/opt/alluxio/bin/alluxio fs load $1 --progress` command. There is no way for us to know how long the load will run, so we shouldn't bound the overall time.
By using go channel we can abort the stall command